sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnarToRowExec.scala

```diff
@@ -39,7 +39,8 @@ class AcceleratedColumnarToRowIterator(
     batches: Iterator[ColumnarBatch],
     numInputBatches: GpuMetric,
     numOutputRows: GpuMetric,
-    totalTime: GpuMetric) extends Iterator[InternalRow] with Arm with Serializable {
+    totalTime: GpuMetric,
+    totalTimeInclusive: GpuMetric) extends Iterator[InternalRow] with Arm with Serializable {
```

Comment:
Sure. I think in most cases we are only interested in the processing time within each operator, exclusive of the time it takes to execute the child operators. Sometimes we do measure `fetchTime` as well though. 

In this case, I needed a way to see the total cost of the transition and everything beneath it so that I could calculate the net cost of the `WholeStageCodegenExec` section. There are some other options for doing this:

1. Update Spark to capture metrics for `InputAdapter`. This would be the most natural but obviously, will only work with newer Spark versions and I don't know how willing the community would be in accepting this new feature. 
2. Create a new operator that just captures time metrics for its child and insert that below the `InputAdapter` when profiling.
3. Add a flag to `GpuColumnarToRow` so that we only create this new `totalTimeInclusive` metric when running the optimizer in this mode, since the metric has little value otherwise.

Reply:
I am more concerned with the naming of the metric and what we can/should be doing long term.

@jlowe @tgravescs do either of you have an opinion on this? 

I am inclined to drop TOTAL_TIME and instead replace it with something more like a GPU_OP_TIME, that computes the time on the GPU that this operator was active for (not something that would be done as a part of this issue).  And then have a FETCH_TIME metric that we could use to indicate the amount of time that this operator spent fetching the data. It would really only be used for operations that did a transition to/from the GPU and possibly for joins where we have to worry about fetching from multiple locations. Then TOTAL_TIME_INCLUSIVE would be the sum of GPU_OP_TIME and FETCH_TIME.

Reply:
I agree "total time" has been a recurring pain point with users since they don't know quite what it is measuring.  Clearly we need a metric for measuring the isolated GPU execution time.  For some nodes, like input nodes or nodes that collect from shuffle like the coalesces, it's interesting to measure the time spent fetching from the input iterator.  I think I would lean towards just reporting those two metrics and not reporting the total time.  Total time tends to lead users to blame that node when the metric is large despite often the problem occurring somewhere upstream to that node in the stage.

Reply:
Sounds good. @andygrove that much of a change feels too large to do here for all of the operators.  But could you do it for the ones that you care about, and then we can file a follow on issue to do it for the others?

Reply:
Unless anyone disagrees, I don't think we need to try and get this PR into 0.5 so I will create a separate PR for the metrics change for the two transition operators (`GpuColumnarToRow` and `GpuRowToColumnar`) and then rebase this PR when that one is merged.